server/src/main/java/org/elasticsearch/index/engine/LuceneChangesSnapshot.java

```diff
@@ -55,6 +53,7 @@
  * A {@link Translog.Snapshot} from changes in a Lucene index
  */
 final class LuceneChangesSnapshot implements Translog.Snapshot {
+    static final int SEARCH_BATCH_SIZE = 100;
```

Comment:
Can this be specified as a parameter via `newLuceneChangesSnapshot(...)` method instead of a constant? Then in ccr we can make this configurable.
 
Also 100 feels on the low side to me. Maybe default to `1024`?